octoprint/datadog_checks/octoprint/check.py

```diff
@@ -0,0 +1,138 @@
+# OctoPrint API key as an instance value like so:
+# ```
+# instances:
+#   - octo_api_key: APIKEYGOESHERE
+# ```
+
+# TODO:
+# - Refactor Octo HTTP API calls
+
+import logging
+import os
+
+import requests
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.utils.subprocess_output import get_subprocess_output
+
+__version__ = "0.1.2"
+
+TIMEOUT = 10
+
+def __init__():
+    logger = logging.getLogger('{}.{}'.format(__name__, self.name))
+    self.log = CheckLoggingAdapter(logger, self)
+    hostname = os.popen("hostname").readline().strip()
+    self.log.debug('hostname: %s', hostname)
+    SERVER = "http://" + hostname
+    self.log.debug('OctoSERVER: %s', SERVER)
+
+class OctoPrintCheck(AgentCheck):
+    def get_rpi_core_temp(self):
+        if os.path.isfile("/usr/bin/vcgencmd"):
+            temp, err, retcode = get_subprocess_output(
+                ["/usr/bin/vcgencmd", "measure_temp"], self.log, raise_on_empty_output=True
+            )
+            self.log.debug('rpi core temp - temp: %s', temp)
+            self.log.debug('rpi core temp - err: %s', err)
+            self.log.debug('rpi core temp - retcode: %s', retcode)
+            temp = temp.replace("temp=", "").replace("'C", "")
+            if temp.startswith("VCHI initialization failed"):
+                self.log.info(
+                    "Unable to get rPi temp.  To resolve, add the 'video' group to the 'dd-agent' user"
+                    " by running `sudo usermod -aG video dd-agent`"
+                )
+                temp = 0.0
+        else:
+            self.log.info(
+                "The command typically used to get the core temperature, /usr/bin/vcgencmd,"
+                "is not available on this system."
+            )
+            temp = 0.0
+        return float(temp)
+
+    def seconds_to_minutes(self, seconds):
+        if not seconds:
+            return 0
+        else:
+            return int(seconds / 60)
+
+    # Get stats from REST API as json
+    def get_job_info(self, server, key, timeout):
+        PATH = "/api/job"
+        url = server + PATH
+        headers = {"X-Api-Key": key, "content-type": "application/json"}
+        req = requests.get(url, timeout=timeout, headers=headers)
+        return req.json()
+
+    def get_tool_temp(self, server, key, timeout):
+        PATH = "/api/printer/tool"
+        url = server + PATH
+        headers = {"X-Api-Key": key, "content-type": "application/json"}
+        req = requests.get(url, timeout=timeout, headers=headers)
+        return req.json()
+
+    def get_bed_temp(self, server, key, timeout):
+        PATH = "/api/printer/bed"
+        url = server + PATH
+        headers = {"X-Api-Key": key, "content-type": "application/json"}
+        req = requests.get(url, timeout=timeout, headers=headers)
+        return req.json()
+
+    def check(self, instance):
+        octo_api_key = instance.get('octo_api_key')
+        self.log.debug('octo_api_key (from config): %s', octo_api_key)
```

Comment:
Better to move these lines to the `__init__` method added in a previous comment.

Reply:
👍